vars/handlePipelineStepErrors.groovy

```diff
@@ -77,7 +78,11 @@ void call(Map parameters = [:], body) {
         if (config.echoDetails)
             message += formatErrorMessage(config, ex)
         writeErrorToInfluxData(config, ex)
-        if (config.failOnError || config.stepName in config.mandatorySteps) {
+
+        boolean isMandatory = config.stepName in config.mandatorySteps
+        DebugReport.instance.storeStepFailure(config.stepName, ex, config.isResilient && !isMandatory)
```

Comment:
where does this come from `config.isResilient`?

Resilience is handled through `config.failOnError` in this step.

Reply:
Good question. My assumption is that "isResilient" may be present in SDK Pipeline step config. The SDK Pipeline has "handleStepErrors()" and wraps pretty much every step's closure in it. This leads to the situation where error may be handled by handlePipelineStepErrors() in case SDK steps call Piper steps, and by handleStepErrors() for SDK pipeline code. This seems fragile to me, but I don't know enough about it. I also don't know whether "isResilient" would be visible here through parameters when a Piper step is called from within an SDK pipeline step.

Reply:
We have searched for any project internal or external which may be using 'isResilient' in their step configuration, and nothing came up. We have decided to remove this from the SDK pipeline. The actual behavior of the pipeline does not necessarily match the intention behind 'isResilient'. For now, it makes sense that the DebugReport only records whether the exception caused the pipeline to abort.